fix(curriculum): simplify the Remove Items Using splice() challenge without using reduce

RandellDawson opened the pull request:
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

After reading [this forum topic](https://www.freecodecamp.org/forum/t/understanding-the-reduce-method-in-splice-challenge/328194) regarding the use of the `reduce` method in the challenge seed code, I realized we should remove this from the code as the `reduce` method is not introduced until the `Functional Programming` section.   I kept the same idea behind the original challenge, but removed the function.

**NOTE:**  If this PR gets merged, the [corresponding forum topic](https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-remove-items-using-splice/301166) hints/solution will need to be updated as well.

RandellDawson committed:
fix: simplify challenge without using reduce

ojeytonwilliams commented:
@RandellDawson how about starting with
```js
const arr = [2, 5, 1, 1, 5, 2, 1];
```
so the user has use new arguments for splice, rather than just copying one of the examples?

RandellDawson committed:
fix: prevent user from adding new elements to sum to 10

RandellDawson commented:
@ojeytonwilliams I modified the array so the splice arguments would be different than the example.  I also added a test to prevent someone from deleting all the elements and just adding `10` back in (like below).
```js
const arr = [2, 4, 5, 1, 7, 5, 2, 1];
arr.splice(0, 20, 10);
```

ojeytonwilliams reviewed (approved):
LGTM :+1:

Manish-Giri commented on curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/remove-items-using-splice.english.md:
```diff
@@ -31,19 +31,24 @@ let newArray = array.splice(3, 2);
 <section id='instructions'>
-We've defined a function, <code>sumOfTen</code>, which takes an array as an argument and returns the sum of that array's elements. Modify the function, using <code>splice()</code>, so that it returns a value of <code>10</code>.
+
+We've initialized an array `arr`. Use `splice()` to remove elements `arr`, so that it only contains elements that sum to the value of <code>10</code>.
```
```suggestion
We've initialized an array `arr`. Use `splice()` to remove elements from`arr`, so that it only contains elements that sum to the value of <code>10</code>.
```

RandellDawson replied:
Thanks for catching that.

RandellDawson committed:
fix:  add from

Co-Authored-By: Manish Giri <manish.giri.me@gmail.com>

Manish-Giri reviewed (changes requested):
Missing "from"/"in" before `arr`.

Also, something I noticed -
>
```
 let newArray = array.splice(3, 2);
// newArray equals ['really', 'happy']
```

Calling it `newArray` gives the impression that the original array `array` has been modified to this. Maybe `removedArray` or `deletedArray` would be better?

RandellDawson commented:
> Missing "from"/"in" before arr.

I am sorry but I do not understand what you want fixed with the above comment.

Manish-Giri commented:
> I am sorry but I do not understand what you want fixed with the above comment.

You already fixed that here- https://github.com/freeCodeCamp/freeCodeCamp/pull/37829/commits/cfa48dd48107b1166b92815b9a970076216616f1

RandellDawson commented:
@Manish-Giri I was going to add the change to the example code too before it got merged.  You want to make a PR for that?
